codec-redis/src/main/java/io/netty/handler/codec/redis/ConstantBulkStringRedisMessage.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.util.internal.StringUtil;
+
+/**
+ * Constant {@link BulkStringRedisMessage}. This is used for immutable (null or empty).
+ */
+public class ConstantBulkStringRedisMessage implements BulkStringRedisMessage {
```

Comment:
Then, How about merge `ConstantBulkStringRedisMessage` and `RefCountedBulkStringRedisMessage` to just one `BulkStringRedisMessage` and `BulkStringRedisMessage` extend `ByteBufHolder`? I think we can use `EMPTY_BUFFER` for `null` bulk string and `empty` bulk string like `LastHttpContent`.
https://github.com/netty/netty/blob/43ebbc3fa065155fa67732b0cbd7c12843b0f3f7/codec-http/src/main/java/io/netty/handler/codec/http/LastHttpContent.java#L30